src/diffusers/utils/hub_utils.py

```diff
@@ -393,6 +406,89 @@ def _get_model_file(
             )
 
 
+# Adapted from
+# https://github.com/huggingface/transformers/blob/1360801a69c0b169e3efdbb0cd05d9a0e72bfb70/src/transformers/utils/hub.py#L976
+# Differences are in parallelization of shard downloads and checking if shards are present.
+
+
+def _get_checkpoint_shard_files(
+    pretrained_model_name_or_path,
+    index_filename,
+    cache_dir=None,
+    force_download=False,
+    proxies=None,
+    resume_download=False,
+    local_files_only=False,
+    token=None,
+    user_agent=None,
+    revision=None,
+    subfolder="",
+    commit_hash=None,
```

Comment:
(nit) 
1. would be nice to have type annotations here even if it's an internal helper. For example, `subfolder` kwarg accepts `None` is most cases in HF libs but not here, so better to mention it explicitly.
2. If used only internally and always with all parameters, better to not set default values. This way if you forget to forward a value when calling it, linter will let you know (and worse case you have a error when testing it).

Reply:
I would prefer not because of the nature of these functions. They can be changed in the future so, I think it’s okay this way.

Reply:
I'm not sure to understand why the nature of these functions makes it better to set default values (since they are not used). But agree it's not critical anyway

Reply:
FWIW, I have removed the arguments that were not being used (`commit_hash`, for example).